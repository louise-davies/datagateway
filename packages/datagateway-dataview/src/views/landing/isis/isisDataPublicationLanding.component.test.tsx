import * as React from 'react';
import { initialState as dgDataViewInitialState } from '../../../state/reducers/dgdataview.reducer';
import configureStore from 'redux-mock-store';
import { StateType } from '../../../state/app.types';
import {
  dGCommonInitialState,
  useDataPublication,
  useDataPublications,
} from 'datagateway-common';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { createMemoryHistory, History } from 'history';
import { QueryClient, QueryClientProvider } from 'react-query';
import { generatePath, Router } from 'react-router-dom';
import { render, type RenderResult, screen } from '@testing-library/react';
import { UserEvent } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';
import ISISDataPublicationLanding from './isisDataPublicationLanding.component';
import { paths } from '../../../page/pageContainer.component';

jest.mock('datagateway-common', () => {
  const originalModule = jest.requireActual('datagateway-common');

  return {
    __esModule: true,
    ...originalModule,
    useDataPublication: jest.fn(),
    useDataPublications: jest.fn(),
  };
});

describe('ISIS Data Publication Landing page', () => {
  const mockStore = configureStore([thunk]);
  let state: StateType;
  let history: History;
  let user: UserEvent;

  const renderComponent = (): RenderResult =>
    render(
      <Provider store={mockStore(state)}>
        <Router history={history}>
          <QueryClientProvider client={new QueryClient()}>
            <ISISDataPublicationLanding dataPublicationId="5" />
          </QueryClientProvider>
        </Router>
      </Provider>
    );

  const users = [
    {
      id: 1,
      contributorType: 'principal_experimenter',
      fullName: 'John Smith',
    },
    {
      id: 2,
      contributorType: 'local_contact',
      fullName: 'Jane Smith',
    },
    {
      id: 3,
      contributorType: 'experimenter',
      fullName: 'Jesse Smith',
    },
    {
      id: 6,
      contributorType: 'experimenter',
      fullName: '',
    },
  ];

  const investigationInstrument = [
    {
      id: 1,
      instrument: {
        id: 4,
        name: 'LARMOR',
      },
    },
  ];

  const investigation = {
    id: 1,
    title: 'Title 1',
    name: 'Name 1',
    summary: 'foo bar',
    visitId: '1',
    doi: 'investigation doi 1.1',
    size: 1,
    investigationInstruments: investigationInstrument,
    startDate: '2019-06-10',
    endDate: '2019-06-11',
    releaseDate: '2019-06-12',
  };

  const investigation2 = {
    id: 2,
    title: 'Title 2',
    name: 'Name 2',
    summary: 'foo bar',
    visitId: '2',
    doi: 'investigation doi 1.2',
    size: 2,
    investigationInstruments: investigationInstrument,
    startDate: '2019-06-10',
    endDate: '2019-06-11',
  };

  const initialStudyDataPublicationData = {
    id: 5,
    pid: 'doi 1',
    users: users,
    publicationDate: '2019-06-10',
  };
  const initialInvestigationDataPublicationsData = [
    {
      id: 8,
      pid: 'investigation doi 1.1',
      title: 'Title 1',
      description: 'foo bar',
      content: {
        dataCollectionInvestigations: [
          {
            investigation: investigation,
          },
        ],
      },
      startDate: '2019-06-10',
      endDate: '2019-06-11',
    },
    {
      id: 9,
      pid: 'investigation doi 1.2',
      description: 'foo bar 2',
      content: {
        dataCollectionInvestigations: [
          {
            investigation: investigation2,
          },
        ],
      },
      startDate: '2019-06-10',
      endDate: '2019-06-11',
    },
  ];

  beforeEach(() => {
    state = JSON.parse(
      JSON.stringify({
        dgdataview: dgDataViewInitialState,
        dgcommon: dGCommonInitialState,
      })
    );

    history = createMemoryHistory({
      initialEntries: [
        generatePath(
          paths.dataPublications.landing.isisDataPublicationLanding,
          {
            instrumentId: '4',
            dataPublicationId: '5',
          }
        ),
      ],
    });
    user = userEvent.setup();

    (useDataPublication as jest.Mock).mockReturnValue({
      data: initialStudyDataPublicationData,
    });

    (useDataPublications as jest.Mock).mockReturnValue({
      data: initialInvestigationDataPublicationsData,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('links to the correct url in the datafiles tab', () => {
    it('in normal view', async () => {
      renderComponent();

      await user.click(
        screen.getByRole('tab', {
          name: 'datapublications.details.investigations',
        })
      );

      expect(history.location.pathname).toBe(
        '/browseDataPublications/instrument/4/dataPublication/5/investigation'
      );
    });

    it('in cards view', async () => {
      history.replace({ search: '?view=card' });
      renderComponent();

      await user.click(
        screen.getByRole('tab', {
          name: 'datapublications.details.investigations',
        })
      );

      expect(history.location.pathname).toBe(
        '/browseDataPublications/instrument/4/dataPublication/5/investigation'
      );
      expect(history.location.search).toBe('?view=card');
    });
  });

  it('renders correctly', async () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders structured data correctly', async () => {
    renderComponent();

    expect(
      await screen.findByTestId('landing-dataPublication-user-0')
    ).toBeInTheDocument();

    expect(document.getElementById('dataPublication-5')).toMatchSnapshot();
  });
});
