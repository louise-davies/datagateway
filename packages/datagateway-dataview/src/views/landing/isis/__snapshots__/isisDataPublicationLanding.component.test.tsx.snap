// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS Data Publication Landing page renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-r95e1t-MuiPaper-root"
    data-testid="isis-dataPublication-landing"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-bctfgv-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-60acjh-MuiPaper-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-auto css-q12ytc-MuiGrid-root"
            >
              <img
                alt="STFC Logo"
                src="stfc-logo-white-text.png"
                style="height: auto; max-height: 90px; margin: auto;"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-true css-lc3k14-MuiGrid-root"
            >
              <div
                style="margin: auto;"
              >
                <h4
                  aria-label="branding-title"
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-2iter1-MuiTypography-root"
                >
                  doi_constants.branding.title
                </h4>
                <p
                  aria-label="branding-body"
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-ajcpbw-MuiTypography-root"
                >
                  doi_constants.branding.body
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 css-jdw6jq-MuiPaper-root"
        >
          <div
            class="MuiTabs-root css-1ujnqem-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="datapublication-details-panel"
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                  id="dataPublication-details-tab"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  datapublications.details.label
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-1a4cg4j-MuiButtonBase-root-MuiTab-root"
                  id="datapublication-investigations-tab"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  datapublications.details.investigations
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-11yukd5-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
        id="datapublication-details-panel"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-o9wqf0-MuiTypography-root"
            data-testid="landing-investigation-title"
          >
            Title 1
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            data-testid="landing-datapublication-description"
          >
            foo bar
          </p>
          <div>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-3w6woi-MuiTypography-root"
              data-testid="landing-dataPublication-users-label"
            >
              datapublications.details.users
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="landing-dataPublication-user-0"
            >
              <b>
                Principal Investigator:
              </b>
               John Smith
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="landing-dataPublication-user-1"
            >
              <b>
                Local Contact:
              </b>
               Jane Smith
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="landing-dataPublication-user-2"
            >
              <b>
                Experimenter:
              </b>
               Jesse Smith
            </p>
          </div>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-3w6woi-MuiTypography-root"
            data-testid="landing-dataPublication-publisher-label"
          >
            datapublications.details.publisher
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            data-testid="landing-dataPublication-publisher"
          >
            doi_constants.publisher.name
          </p>
          <div
            class="tour-dataview-citation-formatter MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-3w6woi-MuiTypography-root"
              data-testid="citation-formatter-title"
            >
              datapublications.details.citation_formatter.label
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              data-testid="citation-formatter-details"
            >
              datapublications.details.citation_formatter.details datapublications.details.citation_formatter.details_select_format
            </p>
            <div
              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
              id="citation-formatter"
            >
              <div
                style="display: flex; align-items: center;"
              >
                <div
                  aria-label="datapublications.details.citation_formatter.select_arialabel"
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-1sxr8zo-MuiInputBase-root-MuiInput-root-MuiSelect-root"
                >
                  <div
                    aria-describedby="citation-formatter-error-message"
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                    role="button"
                    tabindex="0"
                  >
                    datapublications.details.citation_formatter.default_format
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="default"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-10xetod-MuiCircularProgress-root"
                  data-testid="loading-spinner"
                  role="progressbar"
                  style="width: 24px; height: 24px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              <i
                data-testid="citation-formatter-citation"
              />
            </p>
            <button
              aria-label="datapublications.details.citation_formatter.copy_citation_arialabel"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
              disabled=""
              id="citation-formatter-copy-citation"
              tabindex="-1"
              type="button"
            >
              datapublications.details.citation_formatter.copy_citation
            </button>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical css-ftc7wc-MuiDivider-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-5 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 MuiGrid-grid-xl-2 css-bcwqfu-MuiGrid-root"
        >
          <div
            class="css-1jzjbfr"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="PublicIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
                />
              </svg>
              datapublications.pid:
            </p>
            <p
              aria-label="doi 1"
              class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
              data-mui-internal-clone-element="true"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                data-testid="landing-dataPublication-pid-link"
                href="https://doi.org/doi 1"
              >
                doi 1
              </a>
            </p>
          </div>
          <div
            class="css-1jzjbfr"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="StorageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
                />
              </svg>
              datapublications.details.format:
            </p>
            <p
              aria-label="doi_constants.distribution.format"
              class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
              data-mui-internal-clone-element="true"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                href="http://www.isis.stfc.ac.uk/groups/computing/isis-raw-file-format11200.html"
              >
                doi_constants.distribution.format
              </a>
            </p>
          </div>
          <div
            class="css-1jzjbfr"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                data-testid="CalendarTodayIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                />
              </svg>
              datapublications.publication_date:
            </p>
            <p
              aria-label="2019-06-10"
              class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
              data-mui-internal-clone-element="true"
            >
              2019-06-10
            </p>
          </div>
          <div
            class="MuiBox-root css-1xlzx9v"
          >
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-10vnzm9-MuiTypography-root"
                data-testid="landing-datapublication-part-label"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="/browseDataPublications/instrument/4/dataPublication/5/investigation/8"
                >
                  Part DOI: investigation doi 1.1
                </a>
              </h6>
              <div
                class="css-1jzjbfr"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                    data-testid="AssessmentIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
                    />
                  </svg>
                  investigations.instrument:
                </p>
                <p
                  aria-label="LARMOR"
                  class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
                  data-mui-internal-clone-element="true"
                >
                  LARMOR
                </p>
              </div>
              <div
                class="css-1jzjbfr"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                    data-testid="CalendarTodayIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                    />
                  </svg>
                  investigations.release_date:
                </p>
                <p
                  aria-label=""
                  class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
                  data-mui-internal-clone-element="true"
                />
              </div>
              <div
                class="css-fm09c2"
              >
                <span
                  aria-label="buttons.cart_loading_failed_tooltip"
                  class=""
                  data-mui-internal-clone-element="true"
                  style="display: inherit;"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation tour-dataview-add-to-cart css-zddlty-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    id="add-to-cart-btn-investigation-1"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddCircleOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                        />
                      </svg>
                    </span>
                    buttons.add_to_cart
                  </button>
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1xlzx9v"
          >
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div>
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-10vnzm9-MuiTypography-root"
                data-testid="landing-datapublication-part-label"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  href="/browseDataPublications/instrument/4/dataPublication/5/investigation/9"
                >
                  Part DOI: investigation doi 1.2
                </a>
              </h6>
              <div
                class="css-1jzjbfr"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                    data-testid="AssessmentIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
                    />
                  </svg>
                  investigations.instrument:
                </p>
                <p
                  aria-label="LARMOR"
                  class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
                  data-mui-internal-clone-element="true"
                >
                  LARMOR
                </p>
              </div>
              <div
                class="css-1jzjbfr"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pg21ai-MuiTypography-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1eayuh3-MuiSvgIcon-root"
                    data-testid="CalendarTodayIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                    />
                  </svg>
                  investigations.release_date:
                </p>
                <p
                  aria-label=""
                  class="MuiTypography-root MuiTypography-body1  css-1a1wuay-MuiTypography-root"
                  data-mui-internal-clone-element="true"
                />
              </div>
              <div
                class="css-fm09c2"
              >
                <span
                  aria-label="buttons.cart_loading_failed_tooltip"
                  class=""
                  data-mui-internal-clone-element="true"
                  style="display: inherit;"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation tour-dataview-add-to-cart css-zddlty-MuiButtonBase-root-MuiButton-root"
                    disabled=""
                    id="add-to-cart-btn-investigation-2"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddCircleOutlineOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                        />
                      </svg>
                    </span>
                    buttons.add_to_cart
                  </button>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ISIS Data Publication Landing page renders structured data correctly 1`] = `
<script
  id="dataPublication-5"
  type="application/ld+json"
>
  {"@context":"http://schema.org","@type":"Dataset","@id":"https://doi.org/doi 1","url":"https://doi.org/doi 1","identifier":"doi 1","name":"Title 1","description":"foo bar","keywords":"doi_constants.keywords","publisher":{"@type":"Organization","url":"doi_constants.publisher.url","name":"doi_constants.publisher.name","logo":"doi_constants.publisher.logo","contactPoint":{"@type":"ContactPoint","contactType":"customer service","email":"doi_constants.publisher.email","url":"doi_constants.publisher.url"}},"creator":[{"@type":"Person","name":"John Smith"},{"@type":"Person","name":"Jane Smith"},{"@type":"Person","name":"Jesse Smith"}],"includedInDataCatalog":{"@type":"DataCatalog","url":"doi_constants.distribution.content_url"},"license":"doi_constants.distribution.license"}
</script>
`;
